trimesh/smoothing.py

```diff
@@ -0,0 +1,106 @@
+            laplacian operator via def laplacian_calculation
+      Output:
+            Smoothes the input mesh (mofidies)
+      """
+      
+      ## Number of passes
+      for _ in range(n_filter):
+            mesh.vertices  = mesh.vertices + lamb*( coo_matrix.dot(laplacian_operator, mesh.vertices) - mesh.vertices)
+
+def filter_humphrey(mesh,alpha,beta,n_filter,laplacian_operator):
+      """
+      Mesh Smoothing using Laplacian algorithm with different weighting schemes
+
+      Articles:
+      J. Vollmer, R. Mencl, and H. Müller; Improved Laplacian Smoothing of Noisy Surface Meshes
+      
+      Input:
+            mesh to be modofied
+            alpha controls the mesh smoothing (shrinkage): 0 not considered, 1 no smoothimg (e.g. alpha=0.1)
+            beta constrols the aggressiveness of smoothing: 0 no smoothimg,   1 full aggressiveness (e.g. beta=0.5)
+            n_filter = (int) (Number of filters passes)
+            laplacian operator via def laplacian_calculation
+      Output:
+            Smoothes the input mesh (mofidies)
+      """
+      
+      # Number of passes
+      vert_o=mesh.vertices.copy();vert_b=mesh.vertices.copy()
+      for _ in range(n_filter):
+            vert_q=mesh.vertices.copy()
+            mesh.vertices=coo_matrix.dot(laplacian_operator, mesh.vertices)
+            vert_b=mesh.vertices-(alpha*vert_o+(1.-alpha)*vert_q)
+            mesh.vertices=mesh.vertices-(beta*vert_b+(1.-beta)*coo_matrix.dot(laplacian_operator, vert_b))
+
+def filter_taubin(mesh,lamb,nu,n_filter,laplacian_operator):
+      """
+      Mesh Smoothing using Laplacian algorithm with different weighting schemes
+
+      Articles:
+      J. Vollmer, R. Mencl, and H. Müller; Improved Laplacian Smoothing of Noisy Surface Meshes
+      
+      Input:
+            mesh to be modofied
+            Lambda and Nu controls the shrinkage and dilate operations, being between 0.  and 1. (e.g. alpha=0.5; nu=0.53)
+            Nu shall be between 0. < 1./lambda-1./nu < 0.1
+            n_filter = (int) (Number of filters passes)
+            laplacian operator via def laplacian_calculation
+      Output:
+            Smoothes the input mesh (mofidies)
+      """
+      
+      # Number of passes
+      for j in range(n_filter):
+            if j%2==0:
+                  mesh.vertices = mesh.vertices + lamb*( coo_matrix.dot(laplacian_operator, mesh.vertices) - mesh.vertices )
+            else:
+                  mesh.vertices = mesh.vertices - nu*( coo_matrix.dot(laplacian_operator, mesh.vertices) - mesh.vertices )              
+
+def laplacian_calculation(mesh,weight_type):
+      """
+      Input: Expects a trimesh mesh
+             weight_type = 1 - Equal Weights
+                         = 2 - Umbrella Weights
+      Output: Laplacian operator (scipy sparse matrix)
+      """
+      vert_neigh=mesh.vertex_neighbors; vert=mesh.vertices; vert_ori=mesh.vertex_normals.copy(); n_vert=len(vert)
```

Comment:
Could we split these out into separate lines without semicolons?

Reply:
Done... Thanks for the suggestion...

Reply:
No prob man, thanks! I didn't say this before, but nice work! This will be very useful!!!